core/node/state_keeper/src/io/persistence.rs

```diff
@@ -482,27 +474,18 @@ mod tests {
         });
 
         let mut batch_result = default_vm_batch_result();
-        batch_result
-            .final_execution_state
-            .deduplicated_storage_log_queries =
-            storage_logs.iter().map(|query| query.log_query).collect();
+        batch_result.final_execution_state.deduplicated_storage_logs = storage_logs.clone();
         batch_result.state_diffs = Some(
             storage_logs
                 .into_iter()
-                .filter(|&log| log.log_type == StorageLogQueryType::InitialWrite)
-                .map(|log| {
-                    let key = StorageKey::new(
-                        AccountTreeId::new(log.log_query.address),
-                        u256_to_h256(log.log_query.key),
-                    );
-                    StateDiffRecord {
-                        address: log.log_query.address,
-                        key: log.log_query.key,
-                        derived_key: key.hashed_key().0,
-                        enumeration_index: 0,
-                        initial_value: log.log_query.read_value,
-                        final_value: log.log_query.written_value,
-                    }
+                .filter(|&log| log.kind == StorageLogKind::InitialWrite)
+                .map(|log| StateDiffRecord {
+                    address: *log.key.address(),
+                    key: h256_to_u256(*log.key.key()),
+                    derived_key: log.key.hashed_key().0,
+                    enumeration_index: 0,
+                    initial_value: U256::zero(), // TODO not sure about this one
```

Comment:
This is the only change that I'm not confident about. This test might actually need an initial value that it now doesn't get anymore.